sdk/spring/spring-cloud-azure-actuator/src/main/java/com/azure/spring/cloud/actuator/storage/StorageBlobHealthIndicator.java

```diff
@@ -35,18 +37,24 @@ public StorageBlobHealthIndicator(BlobServiceAsyncClient blobServiceAsyncClient)
     protected void doHealthCheck(Health.Builder builder) throws Exception {
         if (blobServiceAsyncClient == null) {
             builder.status(NOT_CONFIGURED_STATUS);
-        } else {
-            builder.withDetail(URL_FIELD, blobServiceAsyncClient.getAccountUrl());
-            final Response<BlobServiceProperties> info = blobServiceAsyncClient.getPropertiesWithResponse()
-                                                                               .block(timeout);
-            if (info != null) {
-                builder.up();
-            }
+            return;
         }
+
+        BlobContainerAsyncClient containerAsyncClient = blobServiceAsyncClient.getBlobContainerAsyncClient(
+            NOT_EXISTING_CONTAINER);
+        Response<Boolean> exists = containerAsyncClient.existsWithResponse().block(timeout);
+        Objects.requireNonNull(exists, "Error occurred checking the container existence!");
+
+        builder.up()
+               .withDetail(URL_FIELD, blobServiceAsyncClient.getAccountUrl())
+               .withDetail("containerName", NOT_EXISTING_CONTAINER)
+               .withDetail("exists", exists.getValue())
+               .withDetail("statusCode", exists.getStatusCode());
```

Comment:
can we start by exposing as few as possible details? Let's only keep the URL_FIELD

Reply:
OK. will revert.